src/components/settings/NumberSetting.tsx

```diff
@@ -177,19 +199,21 @@ export const NumberSetting = ({
                         <Button
                             onClick={() => {
                                 setDialogValue(defaultValue);
-                                updateSetting(defaultValue, false);
+                                submit(defaultValue);
                             }}
                             color="primary"
                         >
                             {t('global.button.reset_to_default')}
                         </Button>
                     ) : null}
-                    <Button onClick={closeDialogWithReset} color="primary">
-                        {t('global.button.cancel')}
-                    </Button>
+                    {!liveUpdate ? (
```

Comment:
the `cancel` behaviour can be kept by adding a new state (e.g. `originalValue`) which only gets updated when the change gets persisted, when canceling, the setting gets reset to the `originalValue`

Reply:
I tried something like that, but the problem is at least in the way settings are passed as props, the value passed in initially is updated too. 

I basically had this, but the problem was `originalValue` was being updated too. Is it because I'm not using `useState`?
```
     const closeDialog = useCallback(
        (resetValue: boolean) => {
            setIsDialogOpen(false);

            if (resetValue) {
                setDialogValue(originalValue);
            }
        },
        [originalValue],
    );
```

I'll give it another try with state to see what happens.

Reply:
Ok I figured this out